comms/core/src/peer_manager/peer_storage.rs

```diff
@@ -278,10 +279,41 @@ where DS: KeyValueStore<PeerId, Peer>
         Ok(peers)
     }
 
+    /// Return "good" peers for syncing
+    /// Criteria:
+    ///  - Peer is not banned
+    ///  - Peer has been seen within a defined time span (1 week)
+    ///  - Only returns a maximum number of syncable peers (corresponds with the max possible number of requestable
+    ///    peers to sync)
+    pub fn discovery_syncing(
+        &self,
+        n: usize,
+        excluded_peers: &[NodeId],
+        features: Option<PeerFeatures>,
+    ) -> Result<Vec<Peer>, PeerManagerError> {
+        if n == 0 {
+            return Ok(Vec::new());
+        }
+
+        let query = PeerQuery::new()
+            .select_where(|peer| {
+                features.map(|f| peer.features == f).unwrap_or(true) &&
+                    !peer.is_banned() &&
+                    peer.deleted_at.is_none() &&
+                    peer.last_seen_since().is_some() &&
+                    peer.last_seen_since().expect("Last seen to exist") <=
+                        Duration::from_secs(PEER_ACTIVE_WITHIN_DURATION) &&
+                    !excluded_peers.contains(&peer.node_id)
```

Comment:
Maybe this can be DRYed up into `fn select_if(peer: Peer) -> bool`?
Repetition below.

Reply:
I like the idea 👍🏻